templates/default/backup-diy-vars.sh.erb

```diff
@@ -0,0 +1,71 @@
+#!/bin/bash
+
+#
+#  Dynamically generated by Chef on <%= node["fqdn"] %>
+#  Local modifications will be overwritten by Chef.
+#
+
+# Which database backup script to use (ex: mssql, postgresql, mysql)
+BACKUP_DATABASE_TYPE=<%= @database['type'] %>
+
+# Which filesystem backup script to use (ex: rsync)
+BACKUP_HOME_TYPE=<%= @backup_diy['backup_home_type'] %>
+
+# Which archive backup script to use (ex: tar, tar-gpg)
+BACKUP_ARCHIVE_TYPE=<%= @backup_diy['backup_archive_type'] %>
+
+# Used by the scripts for verbose logging. If not true only errors will be shown.
+STASH_VERBOSE_BACKUP=<%= @backup_diy['verbose'] ? 'TRUE' : 'FALSE' %>
+
+# The base url used to access this stash instance
+STASH_URL=<%= @backup['baseurl'] %>
+
+# The username and password for the user used to make backups (and have this permission)
+STASH_BACKUP_USER=<%= @backup['user'] %>
+STASH_BACKUP_PASS=<%= @backup['password'] %>
+
+# The name of the database used by this instance.
+STASH_DB=<%= @database['name'] %>
+# The path to stash home folder (with trailing /)
+STASH_HOME=<%= File.join(node['stash']['home_path'], "/") %>
+
+# OS level user and group information (typically: 'stash' for both)
+STASH_UID=<%= node['stash']['user'] %>
+STASH_GID=<%= node['stash']['user'] %>
+
+# The path to working folder for the backup
+STASH_BACKUP_ROOT=<%= @backup_diy['temp_path'] %>
+STASH_BACKUP_DB=${STASH_BACKUP_ROOT}/stash-db/
```

Comment:
For mysql, the backup will fail since `mysqldump` will dump to a file but not a directory.

Should we add a condition check?

```erb
<% if node['stash']['database']['type'] == 'mysql' %>
STASH_BACKUP_DB=${STASH_BACKUP_ROOT}/stash-db
<% else %>
STASH_BACKUP_DB=${STASH_BACKUP_ROOT}/stash-db/
<% end %>
```